### PR TITLE
Fixing the build, removing obsolete platforms

### LINKED ISSUE #987: Fix the failing build
AppVeyor build fails, it needs to be fixed.

### PR DESCRIPTION
**Pull request by alexeyzimarev**
Fixes #987
Addresses part of #938 
Merges #842

### CODE REVIEW

File: RestSharp/Authenticators/OAuth/OAuthTools.cs [hunk 1]
```diff
@@ -105,9 +105,9 @@ public static string GetTimestamp(DateTime dateTime)
         /// The set of characters that are unreserved in RFC 2396 but are NOT unreserved in RFC 3986.
         /// </summary>
         /// <seealso cref="http://stackoverflow.com/questions/846487/how-to-get-uri-escapedatastring-to-comply-with-rfc-3986" />
```
Comment (mjwsteenbergen) [inline RestSharp/Authenticators/OAuth/OAuthTools.cs]: As another comment, I agree with these formatting changes, but it kinda feels weird to see them in this PR.

**Comment by mjwsteenbergen** [comment]
As an aside, did you see the new PR template when you created the PR?

**Comment by alexeyzimarev** [comment]
Yes, I have seen the template, it is very good.

Project lock is now removed. We do not merge until it builds. I am still working on it. It builds on my own Appveyor plan, this is why I submitted a PR.

**Comment by alexeyzimarev** [comment]
Now the whole thing just fails to build, what a great things Microsoft made us with this project system :(

**Comment by alexeyzimarev** [comment]
It is still building with msbuild 14 and this is the reason for failure. On my Appveyor account the same code builds fine with msbuld 15. This happens despite `appveyor.yml` has the correct image settings. It might be the Appveyor build configuration on the plan but I cannot even look at it because I don't have permissions. I asked @hallem but got no reply so far. Now I asked Appveyor support why the settings in YAML file is ignored. Will have to wait for their reply.

**Comment by alexeyzimarev** [comment]
These are builds on my Appveyor https://ci.appveyor.com/project/alexeyzimarev/restsharp/history

**Comment by alexeyzimarev** [comment]
Ok, Appveyor people helped me. The build is set up to ignore the `appveyor.yml` file. Everything is taken from the build settings. I guess further work does not make sense until @hallem or @johnsheehan will finally reply to me screaming and do something about this.

**Comment by alexeyzimarev** [comment]
Aha. I think it will use the old appveyor.yml file until we merge it.

**Pull request merged**